watertap/property_models/anaerobic_digestion/adm1_properties.py

```diff
@@ -0,0 +1,455 @@
+                if degrees_of_freedom(blk[k]) != 0:
+                    raise Exception(
+                        "State vars fixed but degrees of freedom "
+                        "for state block is not zero during "
+                        "initialization."
+                    )
+
+        if state_vars_fixed is False:
+            if hold_state is True:
+                return flags
+            else:
+                blk.release_state(flags)
+
+        init_log.info("Initialization Complete.")
+
+    def release_state(blk, flags, outlvl=idaeslog.NOTSET):
+        """
+        Method to release state variables fixed during initialization.
+
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of logging
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="properties")
+
+        if flags is None:
+            return
+        # Unfix state variables
+        revert_state_vars(blk, flags)
+        init_log.info("State Released.")
+
+
+@declare_process_block_class("ADM1StateBlock", block_class=_ADM1StateBlock)
+class ADM1StateBlockData(StateBlockData):
+    """
+    StateBlock for calculating thermophysical properties associated with the ADM1
+    reaction system.
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super().build()
+
+        # Create state variables
+        self.flow_vol = pyo.Var(
+            initialize=1,
+            domain=pyo.NonNegativeReals,
+            doc="Total volumetric flowrate",
+            units=pyo.units.m**3 / pyo.units.s,
+        )
+        self.pressure = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=101325.0,
+            bounds=(1e1, 1e6),
+            doc="Pressure",
+            units=pyo.units.Pa,
+        )
+        self.temperature = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=298.15,
+            bounds=(298.15, 323.15),
+            doc="Temperature",
+            units=pyo.units.K,
+        )
+        self.conc_mass_comp = pyo.Var(
+            self.params.solute_set,
+            domain=pyo.NonNegativeReals,
+            initialize=0.001,
+            doc="Component mass concentrations",
+            units=pyo.units.kg / pyo.units.m**3,
+        )
+        self.anions = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=0.02,
+            doc="Ions in molar concentration",
+            units=pyo.units.kmol / pyo.units.m**3,
+        )
+        self.cations = pyo.Var(
+            domain=pyo.NonNegativeReals,
+            initialize=0.04,
+            doc="Ions in molar concentration",
+            units=pyo.units.kmol / pyo.units.m**3,
+        )
```

Comment:
Unsure why these are separated as `anions` and `cations` instead of using the standard `conc_mol_phase_comp` for molar concentrations. I am guessing there might be reasoning behind it, but we could consider this a point of _potential_ refinement in issue #909